app.py

```diff
@@ -0,0 +1,600 @@
+                                           user['password']):
+                        user_obj = User(user['email'],
+                                        user['username'], user['_id'])
+                        login_user(user_obj)
+                        flash("Logged in successfully", category='success')
+                        return redirect(url_for('builds'))
+                    else:
+                        flash("Incorrect E-mail/Password", category='danger')
+                else:
+                    flash("Incorrect E-mail/Password", category='danger')
+            else:
+                flash("Incorrect E-mail/Password", category='danger')
+    return render_template("login.html")
+
+
+@app.route("/logout")
+@login_required
+def logout():
+    logout_user()
+    flash('Logged out Successfully', category='success')
+    return redirect(url_for('login'))
+
+# CRUD Routes
+
+
+# Create a Record
+@app.route("/build/new", methods=['POST', 'GET'])
+@login_required
+def create_record():
+    exterior = mongo.db.exterior.find()
+    engine = mongo.db.engine.find()
+    running = mongo.db.runninggear.find()
+    interior = mongo.db.interior.find()
+
+    if request.method == 'POST':
+        builds = mongo.db.builds
+
+        users = [str(current_user.email), ]
+
+        record = {
+            'author': current_user._id,
+            'build_name': request.form.get('build_name'),
+            'total': float(request.form.get('total')),
+            'visibility': request.form.get('visibility'),
+            'car': {
+                'make': request.form.get('make'),
+                'model': request.form.get('model'),
+                'trim': request.form.get('trim'),
+                'year': request.form.get('year'),
+                'price': float(request.form.get('price'))
+            },
+            'votes': {
+                'like': {
+                    'count': 1,
+                    'users_liked': users
+                },
+                'dislike': {
+                    'count': 0,
+                    'users_disliked': []
+                }
+            },
+        }
+
+        # Adds exterior collection to record
+        exterior_dict = []
+        for item in exterior:
+            product = request.form.get('exterior_'+item["part_id"]+'_product')
+            if product is not None:
+                exterior_dict.append({
+                    item["part_id"]: {
+                        'product': product,
+                        'link': request.form.get(
+                                'exterior_'+item["part_id"]+'_link'
+                            ),
+                        'price': float(request.form.get(
+                                'exterior_'+item["part_id"]+'_price'
+                            ))
+                    }
+                })
+
+        record.update({'exterior': exterior_dict})
+
+        # Adds Engine collection to record
+        engine_dict = []
+        for item in engine:
```

Comment:
Let's do these repetitive tasks of adding collection to record in a function so it can be abstracted from our route